src/hints/mod.rs

```diff
@@ -213,22 +225,32 @@ impl HintProcessorLogic for SnosHintProcessor {
         hint_data: &Box<dyn core::any::Any>,
         constants: &HashMap<String, Felt252>,
     ) -> Result<HintExtension, HintError> {
-        match self.builtin_hint_proc.execute_hint(vm, exec_scopes, hint_data, constants) {
-            Err(HintError::UnknownHint(_)) => {}
-            res => return res.map(|_| HintExtension::default()),
-        }
-        // First attempt to execute with builtin hint processor
-        let hint_data = hint_data.downcast_ref::<HintProcessorData>().ok_or(HintError::WrongHintData)?;
-        let hint_code = hint_data.code.as_str();
-        if let Some(hint_impl) = self.hints.get(hint_code) {
-            return hint_impl(vm, exec_scopes, &hint_data.ids_data, &hint_data.ap_tracking, constants)
+        if let Some(hpd) = hint_data.downcast_ref::<HintProcessorData>() {
```

Comment:
Could you put the code in each `if let Some(...) = hint_data.downcast_ref::...` in dedicated functions? It would make this clearer.

Reply:
I will make it in the next pr.